apps/mark-scan/frontend/src/pages/jam_cleared_page.tsx

```diff
@@ -0,0 +1,48 @@
+import { assert } from '@votingworks/basics';
+import { AuthStatus } from '@votingworks/types/src/auth/inserted_smart_card_auth';
+import { Main, Screen, Text, H1 } from '@votingworks/ui';
+import { isPollWorkerAuth } from '@votingworks/utils';
+import { useEffect } from 'react';
+import { endCardlessVoterSession } from '../api';
+
+interface Props {
+  authStatus: AuthStatus;
+  stateMachineState: 'jam_cleared' | 'resetting_state_machine_after_jam';
+}
+
+export function JamClearedPage({
+  authStatus,
+  stateMachineState,
+}: Props): JSX.Element {
+  const endCardlessVoterSessionMutation = endCardlessVoterSession.useMutation();
+
+  useEffect(() => {
+    assert(
+      isPollWorkerAuth(authStatus),
+      `Unexpected non-pollworker auth: ${JSON.stringify(authStatus)}`
+    );
+    // End cardless voter session if we haven't already
+    if (
+      stateMachineState === 'resetting_state_machine_after_jam' &&
+      authStatus.cardlessVoterUser &&
+      !endCardlessVoterSessionMutation.isLoading
+    ) {
+      endCardlessVoterSessionMutation.mutate(undefined);
+    }
+  }, [endCardlessVoterSessionMutation, authStatus, stateMachineState]);
```

Comment:
Yeah I went back and forth on this. I don't love doing this from the frontend, but the options I see in the backend are
* introduce the auth package to the state machine, which will expand its scope from just the paper handler device to other business logic
* check for and action on changed state in `app.ts` when we poll status. I think semantically we'd want to change the status query to a mutation like `queryStatusAndMaybeUpdateSession()`. This also feels kind of unwieldy
* add a status poller to the backend `app.ts`?

Am I missing a more logical place to put this in the backend?

Reply:
Yeah I see what you mean - there are some tradeoffs. My thoughts:

- Letting the state machine have auth side effects seems a little weird, I agree. On the other hand, it's job is managing the state of the app, so I think there's an argument to be made that it could also be in charge of auth.
- Adding a status poller to the backend would be fine also. It's an improvement over frontend polling!
- Another idea would be to have the state machine emit state transition events and have a listener for resetting the auth - a little less intensive than polling.

Since we'll be testing this behavior from the backend app interface, I think it's ok to organize things however we need to under the hood - maybe whatever seems the most readable/clear?

Reply:
I think passing the auth service into the state machine would be easier to understand, even with its expansion of scope for the state machine, than adding another mechanism for coordinating the state machine and auth service. As you note, we also get very standard dependency injection with this approach so testing is straightforward and easy understand as well.